### DISCUSSION TITLE
ZEIT deployment

### DISCUSSION THREAD

**Discussion by mamady310** [vercel/next.js#14467] [Help]
I tried to deploy one of my github projects via zeit.co and it failed. But I am not sure exactly why. Below are are some of errors. Theones with the XXX in front were highligted in red: 

14:44:02  
Cloning github.com/mamady310/hotel-rooms (Branch: master, Commit: 9c103f7)
14:44:03  
Cloning completed in 1112ms
14:44:03  
Analyzing source code...
14:44:05  
Installing build runtime...
14:44:05  
Build runtime installed: 521.564ms
14:44:06  
Looking up build cache...
14:44:06  
Build cache not found
14:44:06  
Installing dependencies...
14:44:26  
> core-js@2.6.11 postinstall /vercel/928f988/node_modules/babel-runtime/node_modules/core-js
14:44:26  
> node -e "try{require('./postinstall')}catch(e){}"
14:44:26  
[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m
14:44:26  
[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
14:44:26  
[96m>[94m https://opencollective.com/core-js [0m
14:44:26  
[96m>[94m https://www.patreon.com/zloirock [0m
14:44:26  
[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m
14:44:26  
> core-js@3.6.5 postinstall /vercel/928f988/node_modules/core-js
14:44:26  
> node -e "try{require('./postinstall')}catch(e){}"
14:44:26  
> core-js-pure@3.6.5 postinstall /vercel/928f988/node_modules/core-js-pure
14:44:26  
> node -e "try{require('./postinstall')}catch(e){}"
14:44:28  
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents):
14:44:28  
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
14:44:28  
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
14:44:28  
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
14:44:28  
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/jest-haste-map/node_modules/fsevents):
14:44:28  
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
14:44:28  
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules/fsevents):
14:44:28  
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
14:44:28  
added 1676 packages from 797 contributors in 21.314s
14:44:29  
61 packages are looking for funding
14:44:29  
  run `npm fund` for details
14:44:29  
Running "npm run build"
14:44:29  
> hotel-rooms@0.1.0 build /vercel/928f988
14:44:29  
> react-scripts build
14:44:30  
Creating an optimized production build...
XXX14:44:31  
Failed to compile.
14:44:31  
./src/index.js
14:44:31  
Cannot find module: 'react-router-dom'. Make sure this package is installed.
14:44:31  
You can install this package by running: npm install react-router-dom.
14:44:31  
npm ERR! code ELIFECYCLE
14:44:31  
npm ERR! errno 1
14:44:31  
npm ERR! hotel-rooms@0.1.0 build: `react-scripts build`
14:44:31  
npm ERR! Exit status 1
14:44:31  
npm ERR! 
14:44:31  
npm ERR! Failed at the hotel-rooms@0.1.0 build script.
14:44:31  
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
14:44:31  
npm ERR! A complete log of this run can be found in:
14:44:31  
npm ERR!     /vercel/.npm/_logs/2020-06-22T18_44_31_596Z-debug.log
XXX14:44:31  
Error: Command "npm run build" exited with 1
14:44:33  
Done with "package.json"

**Comment by mamady310** [comment]
I then tried to use netflify and this is the more specific error that I got:


Production: master@HEAD Failed

Failed during stage 'building site': Build script returned non-zero exit code: 1

**Comment by ryanbahan** [comment]
Can you share your project? I did notice this in the logs as well:

```
Creating an optimized production build...
XXX14:44:31
Failed to compile.
14:44:31
./src/index.js
14:44:31
Cannot find module: 'react-router-dom'. Make sure this package is installed.
```

**Reply by mamady310** [reply]
Thank you for your response..Here is the link to my repo..
https://github.com/mamady310/hotel-rooms.git

so far one solution I found was to delete node_modules and package-lock.json and then re-installing..

**Reply by ryanbahan** [reply]
if deleting your packages and reinstalling them works, I'd assume you were just missing a package (possibly react router given the logs above) and that eveything should be fine now. Are you still having any issues after re-installing?

**Reply by mamady310** [reply]
No that did not work. I then went to netlify to deploy and I feel like they gave better errors that I could google 

I don't know if you can read this: 
https://app.netlify.com/sites/zealous-spence-ef7032/deploys/5ef12115e1651e01993aa4cb

A lot of the errors that i am getting metion something about the build command specifically "build command failed" I'm right now trying to figure out where to put this "CI='' npm run build" build command in my react app

**Reply by ryanbahan** [reply]
It looks like you may be missing a React Icons dependency:

```
3:23:37 PM: ./src/components/NavBar.js
3:23:37 PM: Cannot find module: 'react-icons/fi'. Make sure this package is installed.
```

**Reply by mamady310** [reply]
I can try that but I definitely did or I would assume the icons wouldn't appear on my page..its saying that I didn't install other dependencies like react-router-dom but I will do that next....
but this is the common error I get from both netlify and zeit: 

 Command failed with exit code 1: npm run build

**Reply by mamady310** [reply]
Thank you so much for your help :-D

I re-installed react-icons and I also installed some more dependencies that are documented in my github. I then received an email from github showing 4 different links to where my site deployed :-/. I guess from the previous failed attempts.  Then I went to zeit and netlify and see that my website is now deployed..:-/

**Reply by ryanbahan** [reply]
I'm glad you got your stuff deployed! Just for reference, the `Command failed with exit code 1: npm run build` just means that the server tried to run the 'npm run build' command in your package.json, and it failed (because of the missing dependencies), so it exited with code 1 (indicating a failure). Exiting with code 0 would be a success.

**Reply by mamady310** [reply]
but it was there..my package.json file looks the same as the one that didn't deploy so I am trying to get a better understanding of this. 

<img width="785" alt="Screen Shot 2020-06-22 at 6 26 14 PM" src="https://user-images.githubusercontent.com/56899925/85341507-55844200-b4b6-11ea-894f-e572a84b5853.png">
